src/xbot/common/subsystems/pose/BasePoseSubsystem.java

```diff
@@ -0,0 +1,185 @@
+        double deltaX = Math.cos(currentHeading.getValue() * Math.PI / 180) * totalDistance;
+        
+        totalDistanceX.set(totalDistanceX.get() + deltaX);
+        totalDistanceY.set(totalDistanceY.get() + deltaY);
+        
+        // save values for next round
+        previousLeftDistance = currentLeftDistance;
+        previousRightDistance = currentRightDistance;
+    }
+    
+    public ContiguousHeading getCurrentHeading() {
+        updateCurrentHeading();
+        return currentHeading.clone();
+    }
+    
+    public XYPair getFieldOrientedTotalDistanceTraveled() {
+        return getTravelVector().clone();
+    }
+    
+    private XYPair getTravelVector() {
+        return new XYPair(totalDistanceX.get(), totalDistanceY.get());
+    }
+    
+    public XYPair getRobotOrientedTotalDistanceTraveled() {
+        // if we are facing 90 degrees, no change.
+        // if we are facing 0 degrees (right), this rotates left by 90. Makes sense - if you rotate right, you want
+        // your perception of distance traveled to be that you have gone "leftward."
+        return getTravelVector().rotate(-(currentHeading.getValue() - 90)).clone();
+    }
+    
+    public void resetDistanceTraveled() {
+        totalDistanceX.set(0);
+        totalDistanceY.set(0);
+    }
+    
+    public void setCurrentHeading(double headingInDegrees){
+        currentHeading.setValue(headingInDegrees);
+    }
+    
+    /**
+     * This should be called as often as reasonably possible, to increase accuracy
+     * of the "distance traveled" calculation.
+     * 
+     * The PoseSubsystem can't directly own CANTalons, so some command will need to feed in the
+     * distance values coming from the DriveSubsystem. In order to have accurate calculations, these
+     * values need to be in inches, and should never be reset - any resetting should be done here
+     * in the PoseSubsystem
+     */
+    public void updatePose() {
+        updateCurrentHeading();
+        updateDistanceTraveled(getLeftDriveDistance(), getRightDriveDistance());
+    }
+    
+    protected abstract double getLeftDriveDistance();
+    protected abstract double getRightDriveDistance();
+    
+    public double getRobotPitch() {
+        return getRealPitch() - inherentRioPitch.get();
+    }
+    
+    public double getRobotRoll() {
+        return getRealRoll() - inherentRioRoll.get();
+    }
+    
+    private double getRealPitch() {
+        if (rioRotated.get()) {
+            return imu.getRoll();
+        }
+        return imu.getPitch();
+    }
+    
+    private double getRealRoll() {
+        if (rioRotated.get()) {
+            return imu.getPitch();
+        }
+        return imu.getRoll();
+    }
+    
+    public void calibrateInherentRioOrientation() {
+        inherentRioPitch.set(getRealPitch());
+        inherentRioRoll.set(getRealRoll());
+    }
+    
+    @Override
+    public void updatePeriodicData() {
+        updatePose();
```

Comment:
I was just wondering if updatePose should ever be called outside of updatePeriodicData(), maybe just have the latter?